kakao/src/main/kotlin/com/agoda/kakao/Matchers.kt

```diff
@@ -213,4 +220,37 @@ class DrawableMatcher(@DrawableRes private val resId: Int = -1, private val draw
         drawable.draw(canvas)
         return bitmap
     }
+}
+
+
+/**
+ * Matches given background color with the current one
+ *
+ * @param resId Background color resource to be matched (default is -1)
+ * @param colorCode Background color string code to be matched (default is null)
+ */
+class BackgroundColorMatcher(@ColorRes private val resId: Int = -1,
+                             private val colorCode: String? = null) : TypeSafeMatcher<View>() {
+
+    override fun matchesSafely(item: View?): Boolean {
+        if (resId == -1 && Strings.isNullOrEmpty(colorCode)) {
+            return item?.background == null
+        }
+
+        return item?.let{
+            val expectedColor = if (resId != -1) {
+                ContextCompat.getColor(it.context, resId)
+            } else {
+                Color.parseColor(colorCode)
+            }
+
+            it.background != null &&
+            it.background.current is ColorDrawable &&
+            (it.background.current as ColorDrawable).color == expectedColor
+        }?: false
+    }
+
+    override fun describeTo(description: Description) {
+        description.appendText("with background color resource or provided color code")
```

Comment:
You should append here actual values that were passed to the matcher. It makes debugging much easier.